nodejs/aws-infra/experimental/loadBalancer.ts

```diff
@@ -0,0 +1,254 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as docker from "@pulumi/docker";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "../utils";
+
+import * as mod from ".";
+
+export type LoadBalancers = aws.ecs.ServiceArgs["loadBalancers"];
+
+export interface ILoadBalancerProvider {
+    portMappings(containerName: string, name: string, parent: pulumi.Resource): pulumi.Input<aws.ecs.PortMapping[]>;
+    loadBalancers(containerName: string, name: string, parent: pulumi.Resource): LoadBalancers;
+
+    // endpoints(): pulumi.Output<aws.apigateway.x.Endpoint[]>;
+    // defaultEndpoint(): pulumi.Output<aws.apigateway.x.Endpoint>;
+}
+
+export interface PortInfo {
```

Comment:
this is the equivalent aws-infra form of cloud.ContainerPort.  This enables easy porting down to this level.  However, *importantly*  you are not *required* to use this.  We simply provide a lot of heavy lifting if you want to use this form.  however, you can provider your own impl of ILoadBalancerProvider if you want.  And if there are other patterns that are super common, we can provide those as well.  This decoupling allows us to be opinionated, and to provide a lot of value, without locking someone into a particular approach to this sort of thing.